### DISCUSSION TITLE
Adjustments to swim mechanic

### DISCUSSION THREAD

**Discussion by Rusty-Box** [SuperTux/supertux#3304] [Ideas]
Right now, while functional, swimming is quite a new mechanic in the game this in need of fine tuning.

**These are a some of the things I believe would help to improve said mechanic:**
-  When Tux swims into a solid object / tile, he should ricochet of the object / tile relative to his swim speed
- Tux's turning when boosting should be more smooth out a bit and less loose when turning while swimming at normal speed

**Some further additions that would highly improve the swimming experience:**
- Player projectiles should have lower gravity so they shoot in a more straight-ish line, making it much easier to aim and hit a target + porjectiles move a bit slower
- Lower gravity and slower movement speed for all objects + enemies (unless specified otherwise, forr some instances, like Dive Mine or Fish)

**Comment by Rusty-Box** [comment]
Somewhat turning this into a general issue on improving underwater elements in general

**Some further additions that would highly improve the swimming experience:**
- Player projectiles should have lower gravity so they shoot in a more straight-ish line, making it much easier to aim and hit a target + porjectiles move a bit slower
- Lower gravity and slower movement speed for all objects + enemies (unless specified otherwise, forr some instances, like Dive Mine or Fish)

**Comment by Rusty-Box** [comment]
Putting it on high priority, since it is a main mechanic with its own levels that focus on it, thus should be laid on the back end. We need to give it some polish if we want players to enjoy swimming!

**Comment by MatusGuy** [comment]
My.... goodness!!!!
